### PR TITLE
LCI: Malcolm, Alluring Scoundrel + 2 cards

### PR DESCRIPTION
**Pull request by MikeS-NZ**
LCI: Added card scripts and counter support for the following LCI cards and tokens:

- Bonehoard Dracosaur
- Malcolm, Alluring Scoundrel
- Souls of the Lost
- Token: 3/1 red Dinosaur creature

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/bonehoard_dracosaur.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Name:Bonehoard Dracosaur
+ManaCost:3 R R
+Types:Creature Dinosaur Dragon
+PT:5/5
+K:Flying
+K:First strike
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/bonehoard_dracosaur.txt:6]: bad caps

File: forge-gui/res/cardsfolder/upcoming/malcolm_alluring_scoundrel.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ManaCost:1 U
+Types:Legendary Creature Siren Pirate
+PT:2/1
+K:Flash
+K:Flying
+T:Mode$ DamageDone | ValidSource$ Card.Self | ValidTarget$ Player | CombatDamage$ True | Execute$ TrigPutCounter | TriggerZones$ Battlefield | TriggerDescription$ Whenever CARDNAME deals combat damage to a player, put a chorus counter on it. Draw a card, then discard a card. If there are four or more chorus counters on NICKNAME, you may cast the discarded card without paying its mana cost.
+SVar:TrigPutCounter:DB$ PutCounter | Defined$ Self | CounterType$ CHORUS | CounterNum$ 1 | SubAbility$ DBDraw
+SVar:DBDraw:DB$ Draw | NumCards$ 1 | SubAbility$ DBDiscard
+SVar:DBDiscard:DB$ Discard | Defined$ You | NumCards$ 1 | Mode$ TgtChoose | RememberDiscarded$ True | SubAbility$ DBCast
+SVar:DBCast:DB$ Play | Optional$ True | Defined$ Remembered | ConditionPresent$ Card.Self+counters_GE4_CHORUS | DefinedDesc$ the exiled card | ValidZone$ Graveyard | Valid$ Card.IsRemembered+nonLand | WithoutManaCost$ True | IfDesc$ if there are four or more chorus counters on NICKNAME | ForgetPlayed$ True | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/malcolm_alluring_scoundrel.txt:11]: remove nonLand and use `ValidSA$ Spell`

File: forge-gui/res/cardsfolder/upcoming/souls_of_the_lost.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Name:Souls of the Lost
+ManaCost:1 B
+Types:Creature Spirit
+PT:*/1+*
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/souls_of_the_lost.txt:4]: *+1 not 1+*

File: forge-gui/res/tokenscripts/r_3_1_dinosaur.txt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+Name:Dinosaur
```
Comment (Northmoc) [inline forge-gui/res/tokenscripts/r_3_1_dinosaur.txt:1]: Name:Dinosaur Token

File: forge-gui/res/cardsfolder/upcoming/souls_of_the_lost.txt [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+K:AlternateAdditionalCost:Discard<1/Card>:Sac<1/Permanent>
+S:Mode$ Continuous | EffectZone$ All | CharacteristicDefining$ True | SetPower$ X | SetToughness$ Y | Description$ CARDNAME's power is equal to the number of permanent cards in your graveyard and its toughness is equal to that number plus 1.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/souls_of_the_lost.txt:6]: Description$ Fathomless Descent – etc

File: forge-gui/res/cardsfolder/upcoming/bonehoard_dracosaur.txt (hunk 1 above)
Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/bonehoard_dracosaur.txt]: Fixed, thank you!

File: forge-gui/res/cardsfolder/upcoming/malcolm_alluring_scoundrel.txt (hunk 2 above)
Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/malcolm_alluring_scoundrel.txt]: Fixed!

File: forge-gui/res/cardsfolder/upcoming/souls_of_the_lost.txt (hunk 3 above)
Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/souls_of_the_lost.txt]: Thanks for clarifying. I think I used Tarmogoyf's declaration, which is written as `1+*`. I'll keep this in mind

File: forge-gui/res/cardsfolder/upcoming/souls_of_the_lost.txt (hunk 5 above)
Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/souls_of_the_lost.txt]: Added to desc

File: forge-gui/res/tokenscripts/r_3_1_dinosaur.txt (hunk 4 above)
Comment (MikeS-NZ) [reply forge-gui/res/tokenscripts/r_3_1_dinosaur.txt]: Thank you!

File: forge-gui/res/cardsfolder/upcoming/bonehoard_dracosaur.txt [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+SVar:TrigExile:DB$ Dig | Defined$ You | DigNum$ 2 | ChangeNum$ All | DestinationZone$ Exile | RememberChanged$ True | SubAbility$ DBEffectYou
+SVar:DBEffectYou:DB$ Effect | RememberObjects$ RememberedCard | StaticAbilities$ Play | SubAbility$ DBToken | ForgetOnMoved$ Exile
+SVar:Play:Mode$ Continuous | MayPlay$ True | EffectZone$ Command | Affected$ Card.IsRemembered | AffectedZone$ Exile | Description$ You may play the exiled cards this turn.
+SVar:DBToken:DB$ Token | ConditionDefined$ Remembered | ConditionPresent$ Land | TokenScript$ r_3_1_dinosaur | SubAbility$ DBTreasure
+SVar:DBTreasure:DB$ Token | ConditionDefined$ Remembered | ConditionPresent$ Card.nonLand | TokenScript$ c_a_treasure_sac | SubAbility$ DBCleanup
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+DeckHas:Ability$Token|Sacrifice & Type$Treasure|Dinosaur
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/bonehoard_dracosaur.txt:14]: `Type$Artifact|Treasure|Dinosaur`

Comment (MikeS-NZ) [reply forge-gui/res/cardsfolder/upcoming/bonehoard_dracosaur.txt]: Added!

**Review by tool4ever — APPROVED** [review]

**Pull request merged**